### PR TITLE
Closes #1619: Add exit button

### LINKED ISSUE #1619: Add on-screen button to overlay to exit Firefox
### Why/User Benefit/User Problem
Users say it's difficult to exit the app, adding a button to the overlay will help users who can already access the menu overlay to exit the app

### What / Requirements
Design for button to exit the app, on the overlay (This is already part of the IA in https://github.com/mozilla-mobile/firefox-tv/issues/1071#issuecomment-437353066 , and there's a design, we just need the button asset - in the meantime we can use a placeholder icon)
Telemetry to see usage, and measure effectiveness

### Acceptance Criteria (how do I know when I’m done?)
* Button added to overlay that can exit app
* Telemetry probe added to track the number of times the exit button is used

### Design

* Here's the icon https://drive.google.com/open?id=1fGEOhh9Mzv8ABEa4JpqZrH5fHpmp0_Dc&authuser=tshakespeare@mozilla.com
* The icon should be placed to the left of the settings cog

image

### PR DESCRIPTION
**Pull request by dnarcese**
### Pull Request checklist

- [x] This PR includes thorough **tests** or an explanation of why it does not
The only functionality is adding an Android API call, so there is nothing for us to test, imo.
- [x] This PR includes a **CHANGELOG entry** or does not need one

### CODE REVIEW

**Comment by dnarcese** [comment]
@lime124 Would you mind quickly reviewing the UX for the new exit button?
Here is a screenshot:
screen

**Comment by dnarcese** [comment]
# Request for data collection review form

1) What questions will you answer with this data?

- How often do users click the 'exit Firefox' button?

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?

* To understand how users exit the app and better understand user behaviour.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?

- This is the only way to know how often the button is used.

4) Can current instrumentation answer these questions?

- No telemetry currently exists to answer these questions.

5) List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the Mozilla wiki.   

| Measurement Description                        | Data Collection Category | Tracking Bug #                                               |
| ---------------------------------------------- | ------------------------ | ------------------------------------------------------------ |
| When the user clicks the 'Exit Firefox' button | 2                        | [Issue 1619](https://github.com/mozilla-mobile/firefox-tv/issues/1619) |

6) How long will this data be collected?

* @athomazmoz will permanently monitor this data.

7) What populations will you measure?

* All users with telemetry enabled, on release channel.


8) If this data collection is default on, what is the opt-out mechanism for users?

- There is an option in settings to stop sending usage data.

9) Please provide a general description of how you will analyze this data.

- The data will be analyzed through Redash queries and dashboards.

10) Where do you intend to share the results of your analysis?

- On Redash and with mobile teams.

File: app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt [hunk 1]
```diff
@@ -390,6 +391,9 @@ class NavigationOverlayFragment : Fragment(), View.OnClickListener {
             NavigationEvent.PIN_ACTION -> toolbarViewModel.pinButtonClicked()
             NavigationEvent.TURBO -> toolbarViewModel.turboButtonClicked()
             NavigationEvent.DESKTOP_MODE -> toolbarViewModel.desktopModeButtonClicked()
+            NavigationEvent.EXIT_FIREFOX -> {
+                (activity as MainActivity).moveTaskToBack(true)
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt]: Nit: `moveTaskToBack` is an Android method, so we don't need this cast.

File: app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt [hunk 2]
```diff
@@ -392,6 +392,7 @@ class NavigationOverlayFragment : Fragment(), View.OnClickListener {
             NavigationEvent.TURBO -> toolbarViewModel.turboButtonClicked()
             NavigationEvent.DESKTOP_MODE -> toolbarViewModel.desktopModeButtonClicked()
             NavigationEvent.EXIT_FIREFOX -> {
+                toolbarViewModel.exitFirefoxButtonClicked()
                 (activity as MainActivity).moveTaskToBack(true)
             }
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt]: This probably sounds ridiculous given how simple this case is, but simple cases can quickly build up into spaghetti code.

It's important to keep our data flows consistent, and be very strict about only allowing objects to touch things that are their responsibilities.  In this code, the `NavigationOverlayFragment` knows that clicks to the 'exit Firefox' button mean we need to exit the app.  But what if we want to reuse this exit code elsewhere, based on some other input?  Or if we're looking through the `ViewModel` trying to get a quick look at everything the overlay screen does, not knowing there's a _little_ bit of extra logic tucked within the `View`?  The `ViewModel` is handling the telemetry here, which is good, but it should handle all logic for the `View`, even something as simple as this.  

Again, a little silly for something this small, but consistency matters.  I think this code should call the VM and do nothing else.  Then the VM can send an event down the `events` LiveData, which can handle actually closing the app.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -168,4 +168,7 @@
     Clicking on that button in the YouTube app will show YouTube's instructions on how to cast. -->
     <string name="youtube_casting_onboarding_toast">Cast YouTube videos from your phone or tablet. Open YouTube on your device and tap %s to get started.</string>
 
+    <!-- Content description for the Exit Firefox button -->
+    <string name="exit_firefox_a11y">Exit Firefox</string>
```
Comment (liuche) [inline app/src/main/res/values/strings.xml]: Nit: can you pull the "Firefox" name out of this string, similar to the `firefox_tv_brand_name` string, Firefox TV? Could be called `firefox_tv_brand_name_short`.

**Pull request merged**